osquery/tables/networking/aws_metadata.cpp

```diff
@@ -0,0 +1,238 @@
+/*
+ *  Copyright (c) Jean-Francois Dive <jfdive@gmail.com>
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. 
+ *
+ */
+
+#include <string>
+
+#include <boost/network/protocol/http/client.hpp>
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/noncopyable.hpp>
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+
+namespace pt = boost::property_tree;
+namespace http = boost::network::http;
+
+namespace osquery {
+namespace tables {
+
+/**
+ * @brief SuperClass for AWS metadata accessors
+ */
+class AwsData : public boost::noncopyable {
+  protected:
+  /**
+   * @brief SQL type for the value
+   */
+  ColumnType sqlType;
+
+  /**
+   * @brief SQL column name
+   */
+  std::string fieldName;
+
+  /**
+   * @brief API URL
+   */
+  std::string subUrl;
+
+  /**
+   * @brief HTTP get the data
+   *
+   * @return osquery Status
+   */
+  std::string DoGet() const;
+
+  public:
+  /**
+   * @brief ctor
+   */
+  AwsData(ColumnType _sqlType, std::string _fieldName, std::string _subUrl)
+    : sqlType(_sqlType), fieldName(std::move(_fieldName)), subUrl(std::move(_subUrl)) {}
+
+  /**
+   * @brief dtor
+   */
+  virtual ~AwsData() {}
+
+  /**
+   * @brief Extract relevant data from return API call, pure virtual
+   *
+   * @param r The row to which the value need to be added
+   * @param http_body content of the http response body
+   * @return osquery Status
+   */
+  virtual Status ExtractResult(Row& r, std::string http_body) const = 0;
```

Comment:
It looks like `http_body` is an input-only parameter? Do you mind moving the input parameters to the beginning of the method signature? Also, use a `const std::string&` to avoid the unneeded copy.